pkg/cmd/infra/router/template.go

```diff
@@ -54,6 +55,7 @@ type TemplateRouter struct {
 }
 
 func (o *TemplateRouter) Bind(flag *pflag.FlagSet) {
+	flag.StringVar(&o.RouterName, "name", util.Env("ROUTER_SERVICE_NAME", "public"), "The name the router will identify itself with in the route status")
```

Comment:
Not sure what you mean - this isn't defaulting service name, this is reusing the service name as the common override for the router name (the name of the router service should be the router name).